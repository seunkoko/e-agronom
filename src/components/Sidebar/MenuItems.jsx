// react library
import React from 'react';
import { Link } from 'react-router-dom';

// styles
import './Sidebar.scss';

/**
 * Renders MenuItems component
 * @param {object} props - component props
 * @returns {JSX} JSX
 */
const MenuItems = ({
  id,
  name
}) => (
  <Link
    to={`/${id}`}
    className="links"
  >
    <div className="menu-items">
      <i className="fas fa-cat menu-cat-icon"></i>
      <p className="menu-item-name">{name}</p>
    </div>
  </Link>
);

export default MenuItems;
