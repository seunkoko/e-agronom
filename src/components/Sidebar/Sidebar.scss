@import "../../stylesheets/variables";

.sidebar {
  position: fixed;
  background-color: $primary-black;
  height: 100vh;
  color: $primary-white;
  font-weight: bold;
  width: 20%;

  @media (max-width: $laptop-width) {
    width: 25%;
  }

  @media (max-width: $mobile-width) {
    width: 32%;
  }

  @media (max-width: $phone-width) {
    width: 20%;
  }
}

.sidebar-title {
  margin: 30px 20px;

  &:hover {
    cursor: pointer;
    color: $primary-orange;
  }
}

.menu-items{
  cursor: pointer;
  width: 100%;

  &:hover {
    background-color: $hover-orange;
    color: $primary-black;
  }
}

.menu-cat-icon {
  margin-left: 20px;
}

.menu-item-name {
  display: inline-block;
  margin-left: 10px;
}

.open-sidebar {
  cursor: pointer;
  margin: 30px 20px;
  font-size: 30px;
}
