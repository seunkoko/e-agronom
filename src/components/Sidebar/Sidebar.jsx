// react library
import React from 'react';

// components
import MenuItems from './MenuItems';

// styles
import './Sidebar.scss';

/**
 * Renders Sidebar component
 * @param {object} props - component props
 * @returns {JSX} JSX
 */
const Sidebar = ({
  openMobileMenu,
  categories
}) => (
  <div id="sidebar-id" className="sidebar">
    <div className="show-on-phone">
      <i
        id="open-sidebar-icon"
        className="fas fa-arrow-right open-sidebar"
        onClick={openMobileMenu}
      >
      </i>
    </div>

    <div id="sidebar-content" className="hide-on-phone">
      <h2 className="sidebar-title">
        CATEGORIES
      </h2>

      {
        categories && categories.map(category => {
          const item = <MenuItems
            key={category.id}
            id={category.id}
            name={category.name}
          />
          return item;
        })
      }
    </div>
  </div>
);

export default Sidebar;
