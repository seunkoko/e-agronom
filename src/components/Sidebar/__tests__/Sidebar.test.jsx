// react library
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { spy } from 'sinon';

// components
import Sidebar from '../Sidebar';

describe('MainView component', () => {
  let wrapper;
  let props = {
    openMobileMenu: jest.fn(),
    categories: [],
  }

  beforeEach(() => {
    wrapper = shallow(
      <Sidebar { ...props } />
    );
  });

  it('renders correctly', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when categories are provided', () => {
    props = {
      ...props,
      categories: [
        {
          id: '1',
          name: 'category1'
        },
        {
          id: '2',
          name: 'category2'
        },
      ],
    }

    wrapper = shallow(
      <Sidebar { ...props } />
    );
    expect(wrapper).toBeTruthy();
  });
});
