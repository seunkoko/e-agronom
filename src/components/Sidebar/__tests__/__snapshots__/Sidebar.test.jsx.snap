// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView component renders correctly 1`] = `
<div
  className="sidebar"
  id="sidebar-id"
>
  <div
    className="show-on-phone"
  >
    <i
      className="fas fa-arrow-right open-sidebar"
      id="open-sidebar-icon"
      onClick={[MockFunction]}
    />
  </div>
  <div
    className="hide-on-phone"
    id="sidebar-content"
  >
    <h2
      className="sidebar-title"
    >
      CATEGORIES
    </h2>
  </div>
</div>
`;
